libs/@local/hashql/diagnostics/src/severity.rs

```diff
@@ -1,9 +1,37 @@
-use core::{fmt::Display, mem};
+use core::{
+    fmt::{self, Display},
+    mem,
+};
 
 use anstyle::Color;
 
 use crate::{Help, Note};
 
+pub const trait SeverityKind: Copy + const Into<Severity> {
+    /// Returns whether this severity level is fatal.
+    ///
+    /// A severity is considered fatal if its code is >= 400.
+    /// Fatal severities are `Error`, `Fatal`, and `Bug`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use hashql_diagnostics::{Severity, severity::SeverityKind};
+    ///
+    /// // Critical severities
+    /// assert!(Severity::Bug.is_critical());
+    /// assert!(Severity::Fatal.is_critical());
+    /// assert!(Severity::Error.is_critical());
+    ///
+    /// // Non-critical severities
+    /// assert!(!Severity::Warning.is_critical());
+    /// assert!(!Severity::Note.is_critical());
+    /// assert!(!Severity::Debug.is_critical());
+    /// ```
+    fn is_critical(self) -> bool;
+    fn is_advisory(self) -> bool;
```

Comment:
We could move the non-critical examples to `is_advisory`. Also it's worth mentioning that a severity should either be critical OR advisory and that it's the responsibility of the implementor to upkeep this.
Probably not worth it but we could seal the trait.